packages/xdg_directories/lib/xdg_directories.dart

```diff
@@ -0,0 +1,163 @@
+/// `$XDG_CACHE_HOME`).
+///
+/// Throws [StateError] if the HOME environment variable is not set.
+Directory get cacheHome => _directoryFromEnvironment('XDG_CACHE_HOME', '.cache');
+
+/// The list of preference-ordered base directories relative to
+/// which configuration files should be searched. (Corresponds to
+/// `$XDG_CONFIG_DIRS`).
+///
+/// Throws [StateError] if the HOME environment variable is not set.
+List<Directory> get configDirs => _directoryListFromEnvironment('XDG_CONFIG_DIRS', '/etc/xdg');
+
+/// The a single base directory relative to which user-specific
+/// configuration files should be written. (Corresponds to `$XDG_CONFIG_HOME`).
+///
+/// Throws [StateError] if the HOME environment variable is not set.
+Directory get configHome => _directoryFromEnvironment('XDG_CONFIG_HOME', '.config');
+
+/// The list of preference-ordered base directories relative to
+/// which data files should be searched. (Corresponds to `$XDG_DATA_DIRS`).
+///
+/// Throws [StateError] if the HOME environment variable is not set.
+List<Directory> get dataDirs => _directoryListFromEnvironment('XDG_DATA_DIRS', '/usr/local/share:/usr/share');
+
+/// The base directory relative to which user-specific data files should be
+/// written. (Corresponds to `$XDG_DATA_HOME`).
+///
+/// Throws [StateError] if the HOME environment variable is not set.
+Directory get dataHome => _directoryFromEnvironment('XDG_DATA_HOME', '.local/share');
+
+/// The base directory relative to which user-specific runtime
+/// files and other file objects should be placed. (Corresponds to
+/// `$XDG_RUNTIME_DIR`).
+///
+/// Throws [StateError] if the HOME environment variable is not set.
+Directory get runtimeDir => _directoryFromEnvironment('XDG_RUNTIME_DIR', null);
+
+/// Gets the xdg user directory named by `dirName`.
+///
+/// Use [getUserDirectoryNames] to find out the list of available names.
+Directory getUserDirectory(String dirName) {
+  final ProcessResult result = _processManager.runSync(
+    <String>['xdg-user-dir', dirName],
+    includeParentEnvironment: true,
+    stdoutEncoding: Encoding.getByName('utf8'),
+  );
+  final String path = utf8.decode(result.stdout).split('\n')[0];
+  return Directory(path);
+}
+
+/// Gets the set of user directory names that xdg knows about.
+///
+/// These are not paths, they are names of xdg values.  Call [getUserDirectory]
+/// to get the associated directory.
+///
+/// These are the names of the variables in "[configHome]/user-dirs.dirs", with
+/// the `XDG_` prefix removed and the `_DIR` suffix removed.
+Set<String> getUserDirectoryNames() {
+  final File configFile = File(path.join(configHome.path, 'user-dirs.dirs'));
+  List<String> contents;
+  try {
+    contents = configFile.readAsLinesSync();
+  } on FileSystemException catch (e) {
+    return const <String>{};
+  }
+  final Set<String> result = <String>{};
+  final RegExp dirRegExp = RegExp(r'^\s*XDG_(?<dirname>.*)_DIR\s*=\s*(?<dir>.*)\s*$');
```

Comment:
s/`.`/`[^=]`/ maybe? Theoretically, although extremely unlikely, _DIR= could be part of a path value.

Reply:
Good point, thanks.  Added.